### PR TITLE
PX4 motor random spin up issues

### PR DESCRIPTION
**Pull request by kevindehecker**
Turned out that having 1500000 baud between the f1 and f4 on the Pixhawk board was rather heisenbuggy. After some time (ca half an hour stationary, but with power from battery) packages started getting lost. 

To remain backwards compatible with PX4 firmware, the 150000 baud is still required. However, during flight not. So the baud rate is set to 230400 after 10 seconds.

This PR
-fixes the above baud issue, 
-adds some signalling information to the FBW LED,
-fixes some issues related to FBW_MANUAL_MODE , which caused the motors to spin up shortly during intermcu loss.

### CODE REVIEW

File: sw/airborne/firmwares/rotorcraft/main_fbw.c [hunk 1]
```diff
@@ -178,18 +176,29 @@ STATIC_INLINE void main_periodic(void)
...
   }
 
+#ifdef BOARD_PX4IO
+  //due to a baud rate issue on PX4, for a few seconds the baud is 1500000 however this may result in package loss, causing the motors to spin at random
+  //to prevent this situation:
+  if (inter_mcu.stable_px4_baud != PPRZ_BAUD) {
+    fbw_mode = FBW_MODE_FAILSAFE;
+    autopilot_motors_on = false;
+    //signal to user whether fbw can be flashed:
+    LED_OFF(FBW_MODE_LED); // causes really fast blinking
```
Comment (flixr) [inline sw/airborne/firmwares/rotorcraft/main_fbw.c]: please protect this with `#ifdef FBW_MODE_LED`

File: sw/airborne/subsystems/intermcu/intermcu_fbw.c [hunk 2]
```diff
@@ -119,6 +119,19 @@ void intermcu_send_status(uint8_t mode)
   //FIXME
 }
 
+
+void intermcu_blink_fbw_led(uint16_t dv)
+{
+  static uint16_t idv = 0;
+  if (!autopilot_motors_on) {
```
Comment (flixr) [inline sw/airborne/subsystems/intermcu/intermcu_fbw.c]: Please protect this with `#ifdef FBW_MODE_LED`

**Comment by flixr** [comment]
Do I see that correctly that this is currently not usable for the fixedwing firmware?

**Comment by kevindehecker** [comment]
I made and tested it for rotorcraft (for which the manual mode mostly useless, unless an additional IMU is added. This will happen in the near future for some of our airframes). That's why the changes are only in rotorcraft/main_fbw right?

**Comment by flixr** [comment]
We can merge this, just wondering if you also plan to add the stuff required for px4 flashing to the fixedwing intermcu? Since the pixhawk will probably also be used for fixedwings...

**Comment by kevindehecker** [comment]
Ah, I see your point. I guess I can add the `#ifdef BOARD_PX4IO` part from rotorcraft/main_fbw.c to fixedwing, but I wouldn't be able to test it and frankly the fixedwing intermcu code looks quite different (outdated?). I 'm not very enthusiastic about that idea yet. Shall I add a FIXME in the code fixedwing main for now?

To be clear, the flashing part should work just fine. The only problem is that it may switch to FBW_MODE_MANUAL during this first 10s of 150000 baud after powerup. In case of rotorcraft this is a big issue, the motors spin up to minimum rotating thrust. This is I think mostly not the case for fixed wings.

**Comment by agressiva** [comment]
Kevin, i can test the FW firmware.

btw, i tried compile to FW without success.

**Comment by kevindehecker** [comment]
Thx agressiva, but I think there is a long way to go before flight tests :|

It seems intermcu status is not checked at all in fixed wing... There is probably more worked involved offtopic for this PR. I say merge this now.

**Pull request closed without merging**